go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -43,7 +43,7 @@ func (bnc *BaseNetworkController) allocatePodIPs(pod *kapi.Pod,
 func (bnc *BaseNetworkController) allocatePodIPsOnSwitch(pod *kapi.Pod,
 	annotations *util.PodAnnotation, nadName string, switchName string) (expectedLogicalPortName string, err error) {
 
-	if !util.PodScheduled(pod) || util.PodWantsHostNetwork(pod) || util.PodCompleted(pod) {
```

Comment:
can the following happen?
1. we get completed pod event and run delete handler for it
2. completed pod ip is allocated for a new pod
3. ovn-k is restarted and it tries to allocate all existing pod ips (completed pod still exists in the kapi)
4. `allocatePodIPsOnSwitch` is called by `syncPods` for both completed and new pods with the same ip

Reply:
Yes, that can happen. We would log the warning that I changed a few lines below. But those IPs would be attributed and only de-allocated for the new pod, not the completed pod, which is the purpose of the second commit. Let me know if there is any other reason you would think this could be a problem.

Reply:
So this function now may allocate ip for a completed pod and log error for a new pod using the same ip (depending on the pods order in syncPods), but in the following code it will be handled as if the ip was actually allocated for the new pod and not the completed pod?
If so, it may worth updating the comment for this function, otherwise this behaviour may look wrong, especially the part where it logs allocation warning for a running pod (btw it would be nice to distinguish an error from an expected overlap here somehow)

Reply:
Yeah, I can add a comment.

I don't think it is worth it to avoid the warning for an expected overlap in respect to the code changes required. But I guess I can just print the warning for completed pods. WDYT?

Reply:
> I don't think it is worth it to avoid the warning for an expected overlap in respect to the code changes required. But I guess I can just print the warning for completed pods. WDYT?

But now that I think about it, the warning for the completed pod would only happen depending on the order in which we process them. I think it is nice to have a signal that we have multiple pods with the same allocated IP. I can add the pod phase to the warning so we know in which status it was in.

Reply:
yeah I think just adding more info to the code comments and the warning itself may be useful to not cause false-positive concerns around this